actions/src/sema4ai/actions/agent/_client.py

```diff
@@ -0,0 +1,196 @@
+import json
+import logging
+import os
+import platform
+from copy import copy
+from urllib.parse import urljoin, urlparse
+
+import sema4ai_http
+
+log = logging.getLogger(__name__)
+
+
+class _AgentAPIClient:
+    API_ENDPOINT_VERSIONS = ["v1", "v2"]
+    API_PATH_PREFIX = "api/public"
+    ENV_VAR_NAME = "SEMA4AI_AGENTS_SERVICE_URL"
+    PID_FILE_NAME = "agent-server.pid"
+    CONNECTION_TIMEOUT = 1
+
+    def __init__(self, api_key: str | None = None):
+        """Initialize the AgentServerClient."""
+        self.api_url = self._get_api_url()
+        self.api_key = api_key
+        log.info(f"API URL: {self.api_url}")
+
+        # Determine if we're running in cloud environment
+        self.is_cloud = self._is_cloud_environment()
```

Comment:
I don't think you need to have this distiction... rather, if the api_key is passed, always pass the header, otherwise don't.